doc/_extensions/zephyr/gh_utils.py

```diff
@@ -46,12 +47,33 @@
 from sphinx.application import Sphinx
 from sphinx.util.i18n import format_date
 
+ZEPHYR_BASE = Path(__file__).parents[3]
+SCRIPTS = ZEPHYR_BASE / "scripts"
+sys.path.insert(0, str(SCRIPTS))
+from get_maintainer import Maintainers
+
+MAINTAINERS = Maintainers()
+
 
 __version__ = "0.1.0"
 
 
 def get_page_prefix(app: Sphinx, pagename: str) -> str:
-    if not os.path.isfile(app.env.project.doc2path(pagename)):
+    """Return the prefix that needs to be added to the page path to get its location in the
+    repository.
+
+    If pagename refers to a page that is automatically generated by Sphinx or if it matches one of
+    the patterns in ``gh_link_exclude`` configuration option, return None.
+
+    Args:
+        app: Sphinx instance.
+        pagename: Page name (path).
+
+    Returns:
+        Prefix if applicable, None otherwise.
+    """
+
+    if not os.path.isfile(app.env.doc2path(pagename)):
```

Comment:
Does the `os.path` library ensure correctness when working on both Linux and Windows?

Otherwise I'd suggest to use Pathlib (https://docs.python.org/3/library/pathlib.html) which does this (I see that we are already using Pathlib other places in this file). 

The corresponding function would be https://docs.python.org/3/library/pathlib.html#pathlib.Path.is_file